src/helm/clients/vision_language/huggingface_vision2seq_client.py

```diff
@@ -0,0 +1,145 @@
+from threading import Lock
+from typing import Any, Dict, List, Optional
+
+from dataclasses import dataclass
+from transformers import AutoProcessor, AutoModelForVision2Seq
+from transformers.image_utils import load_image
+import torch
+
+from helm.common.cache import CacheConfig
+from helm.common.gpu_utils import get_torch_device_name, is_cuda_available
+from helm.common.hierarchical_logger import hlog, htrack_block
+from helm.common.media_object import TEXT_TYPE
+from helm.common.request import Request, RequestResult, GeneratedOutput, Token
+from helm.common.request import wrap_request_time
+from helm.common.tokenization_request import TokenizationRequest
+from helm.clients.client import CachingClient, generate_uid_for_multimodal_prompt
+from helm.tokenizers.tokenizer import Tokenizer
+
+
+@dataclass(frozen=True)
+class Vision2SeqModelProcessor:
+    """Loaded model and processor."""
+
+    model: AutoModelForVision2Seq
+    processor: AutoProcessor
+
+
+_models_lock: Lock = Lock()
+_models: Dict[str, Optional[Vision2SeqModelProcessor]] = {
+    "HuggingFaceM4/idefics2-8b": None,
```

Comment:
Any reason we need to explicitly declare models here? Most clients do not validate which model names are supported (because the list can change frequently).

Reply:
It was for name validation. Do you think I can just remove this line then?